src/main/java/seedu/address/model/assignment/AssignmentList.java

```diff
@@ -162,123 +163,218 @@ public void generateSchedule(int numDays) {
+
+            int currSize = cumulativeHoursPerDayList.size();
+
+            // Initialise hoursPerDayList
+            for (int i = 0; i < currSize; i++) {
+                hoursPerDayList.add((float) 0);
+            }
 
             // Allocate hours to new days first (if any)
-            int numNewDays = noOfDaysBetween - (hoursPerDayList.size() - 1);
+            int numNewDays = noOfDaysBetween - (currSize - 1);
 
             if (numNewDays != 0) {
-                float hoursToAdd = estHours / numNewDays;
+                float hoursToAdd = hoursToBeAllocated / numNewDays;
                 float hoursAdded = Math.min(currMinAndSecondMin[0], hoursToAdd);
 
                 for (int i = 0; i < numNewDays; i++) {
+                    cumulativeHoursPerDayList.add(hoursAdded);
                     hoursPerDayList.add(hoursAdded);
-                    estHours -= hoursAdded;
+                    hoursToBeAllocated -= hoursAdded;
                 }
-                numHoursToDeadline += hoursAdded;
             }
 
             // Allocate hours to days with the least amount of work allocated in increasing order
-            while (currMinAndSecondMin[0] != currMinAndSecondMin[1]) {
-                int daysWithMinHours = getMinDays(hoursPerDayList, noOfDaysBetween, currMinAndSecondMin[0]);
+            while (currMinAndSecondMin[0] != currMinAndSecondMin[1] && hoursToBeAllocated != 0) {
+                int daysWithMinHours = getMinDays(cumulativeHoursPerDayList, noOfDaysBetween, currMinAndSecondMin[0]);
                 float diffBetweenMinAndSecondMin = currMinAndSecondMin[1] - currMinAndSecondMin[0];
-                float hoursToAdd = estHours / daysWithMinHours;
+                float hoursToAdd = hoursToBeAllocated / daysWithMinHours;
                 float hoursAdded = Math.min(hoursToAdd, diffBetweenMinAndSecondMin);
 
-                numHoursToDeadline += allocateHoursToMinDays(hoursPerDayList, noOfDaysBetween,
-                    hoursAdded, currMinAndSecondMin[0], estHours);
-                currMinAndSecondMin = getMinAndSecondMin(hoursPerDayList, noOfDaysBetween);
+                hoursToBeAllocated -= allocateHoursToMinDays(cumulativeHoursPerDayList, noOfDaysBetween, hoursAdded,
+                    currMinAndSecondMin[0], hoursPerDayList);
+
+                currMinAndSecondMin = getMinAndSecondMin(cumulativeHoursPerDayList, noOfDaysBetween);
             }
 
             // Allocate remaining hours equally across the days up to deadline of assignment (if any)
-            float hoursAdded = estHours / (noOfDaysBetween);
+            float hoursAdded = hoursToBeAllocated / (noOfDaysBetween + 1);
 
-            for (int k = 0; k < Math.min(hoursPerDayList.size(), noOfDaysBetween); k++) {
+            for (int k = 0; k < Math.min(cumulativeHoursPerDayList.size(), noOfDaysBetween + 1); k++) {
+                cumulativeHoursPerDayList.set(k, cumulativeHoursPerDayList.get(k) + hoursAdded);
```

Comment:
What is the diff between cumulativeHoursPerDayList and hoursPerDayList? Maybe you should put a comment for each variable on what it stores

Reply:
okkk!